docs/en/observability/apm/apm-performance-diagnostic.asciidoc

```diff
@@ -0,0 +1,43 @@
+[[apm-performance-diagnostic]]
+=== APM Server performance diagnostic
+
+[[apm-es-backpressure]]
+[float]
+==== Diagnosing backpressure from {es}
+
+When {es} is under excessive load or indexing pressure, APM Server could experience the downstream backpressure when indexing new documents into {es}.
+Most commonly, backpressure from {es} will manifest itself in the form of higher indexing latency and/or rejected requests, which in return could lead APM Server to deny incoming requests.
+As a result APM agents connected to the affected APM Server will suffer from throttling and/or request timeout when shipping APM events. 
+
+To quickly identify possible issues try looking for similar error logs lines in APM Server logs:
+
+[source,json]
+----
+...
+{"log.level":"error","@timestamp":"2024-07-27T23:46:28.529Z","log.origin":{"function":"github.com/elastic/go-docappender/v2.(*Appender).flush","file.name":"v2@v2.2.0/appender.go","file.line":370},"message":"bulk indexing request failed","service.name":"apm-server","error":{"message":"flush failed (429): [429 Too Many Requests]"},"ecs.version":"1.6.0"}
+{"log.level":"error","@timestamp":"2024-07-27T23:55:38.612Z","log.origin":{"function":"github.com/elastic/go-docappender/v2.(*Appender).flush","file.name":"v2@v2.2.0/appender.go","file.line":370},"message":"bulk indexing request failed","service.name":"apm-server","error":{"message":"flush failed (503): [503 Service Unavailable]"},"ecs.version":"1.6.0"}
+...
+----
+
+To gain better insight into APM Server health and performance, consider enabling the monitoring feature by following the steps in <<apm-monitor-apm-self-install,Monitor a Fleet-managed APM Server>>.
+When enabled APM Server will additionally report a set of vital metrics to help you identify any performance degradation.
+
+Pay careful attention to the next metric fields:
+
+* `beats_stats.metrics.libbeat.output.events.active` that represents the number of buffered pending documents waiting for indexing;
+(_if this value is increasing rapidly it indicates {es} backpressure_)
+* `beats_stats.metrics.libbeat.output.events.acked` that represents the number of indexing operations that have completed successfully;
+* `beats_stats.metrics.libbeat.output.events.failed` that represents the number of indexing operations that failed, it includes all failures;
```

Comment:
nit

```suggestion
* `beats_stats.metrics.libbeat.output.events.failed` that represents the number of indexing operations that failed, including all failures;
```

Reply:
@carsonip did I understand correctly that `beats_stats.metrics.libbeat.output.events.failed` is counting the ES _documents_ that failed, rather than _bulk requests that failed?
The suggestion is based on this assumption.

Reply:
Documentation is not super clear 🤔

Reply:
It was also source of an SDH, this one https://github.com/elastic/sdh-apm/issues/1367#issuecomment-2159748910

Reply:
> @carsonip did I understand correctly that beats_stats.metrics.libbeat.output.events.failed is counting the ES documents that failed, rather than _bulk requests that failed?

correct
```
Failed:                 atomic.LoadInt64(&a.docsFailed),
```

Reply:
I used description from the metrics from the code directly, fixed the wording in the doc now, it should be better.